src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorageService.cs

```diff
@@ -19,6 +20,59 @@ internal partial class SQLitePersistentStorageService : AbstractPersistentStorag
 
         private readonly IPersistentStorageFaultInjector _faultInjectorOpt;
 
+        [DllImport("kernel32.dll")]
+        private static extern IntPtr LoadLibrary(string dllToLoad);
+        private static bool _initialized;
+        private static object _gate = new object();
+
+        internal static bool TryInitializeLibraries()
+        {
+            if (_initialized)
+            {
+                return true;
+            }
+                
+            lock (_gate)
+            {
+                if (_initialized)
+                {
+                    return true;
+                }
```

Comment:
i feel like the momeent after checking, we should set _initialized to true, so we never execute this block more than once.

Reply:
Instead of just returning 'true', i feel a better API would be to have it actually have a different bool stating if we actually initialized properly or not.  (Alternatively, use a ```bool?``` where 'null' means 'have not computed', and non-null is the actual value.)